be/src/exec/json_scanner.cpp

```diff
@@ -727,25 +723,31 @@ Status JsonReader::_read_file_broker() {
                             sz, _scanner->_params.json_file_size_limit));
     }
 
-    if (sz > _payload_buffer_capacity) {
+    if (sz + simdjson::SIMDJSON_PADDING > _file_broker_buffer_capacity) {
         // reallocate if needed.
+        // For efficiency reasons, simdjson requires a string with a few bytes (simdjson::SIMDJSON_PADDING) at the end.
+        // Hence, a re-allocation is needed if the space is not enough.
         auto allocated = sz + simdjson::SIMDJSON_PADDING;
-        _payload_buffer.reset(new char[allocated]);
-        _payload_buffer_capacity = allocated;
+        _file_broker_buffer.reset(new char[allocated]);
+        _file_broker_buffer_capacity = allocated;
+        _file_broker_buffer_size = 0;
     }
 
     {
-        SCOPED_RAW_TIMER(&_counter->file_read_ns);
-        auto res = _file->read(reinterpret_cast<void*>(_payload_buffer.get()), sz);
+        auto res = _file->read(reinterpret_cast<void*>(_file_broker_buffer.get()), sz);
         if (!res.ok()) {
             return res.status();
         }
-        _payload_buffer_size = sz;
+        _file_broker_buffer_size = sz;
```

Comment:
is `sz` definitely equal to `res.value()`?

Reply:
Updated. The correct one should be `_file_broker_buffer_size = res.value()`.